qubesadmin/tools/qvm_template.py

```diff
@@ -1494,33 +1517,52 @@ def repolist(args: argparse.Namespace, app: qubesadmin.app.QubesBase) -> None:
         for idx, path in enumerate(args.repo_files):
             src = os.path.abspath(path)
             # Use index as file name in case of collisions
-            dst = os.path.join(reposdir, '%d.repo' % idx)
+            dst = os.path.join(reposdir, f'{idx:d}.repo')
             os.symlink(src, dst)
         conf = dnf.conf.Conf()
         conf.substitutions['releasever'] = args.releasever
         conf.reposdir = reposdir
         base = dnf.Base(conf)
         base.read_all_repos()
-        if args.repoid:
-            base.repos.get_matching('*').disable()
-            for repo in args.repoid:
-                base.repos.get_matching(repo).enable()
-        else:
-            for repo in args.enablerepo:
-                base.repos.get_matching(repo).enable()
-            for repo in args.disablerepo:
-                base.repos.get_matching(repo).disable()
 
-        repos: typing.List[dnf.repo.Repo]
+        # Filter (name, operation) from args.repos
+        enable_repos = []
+        disable_repos = []
+        repoid = []
+        repos: typing.List[dnf.repo.Repo] = []
         if args.repos:
-            repos = []
             for repo in args.repos:
-                repos += list(base.repos.get_matching(repo))
+                name, operation = repo
+                if operation == "repoid":
+                    repoid.append(name)
+                elif operation == "enablerepo":
+                    enable_repos.append(name)
+                elif operation == "disablerepo":
+                    disable_repos.append(name)
+
+            if repoid:
+                if enable_repos or disable_repos:
+                    print("Warning: Ignoring --enablerepo and --disablerepo "
+                          "options.", file=sys.stderr)
+                base.repos.get_matching('*').disable()
+                for repo in repoid:
+                    dnf_repo = base.repos.get_matching(repo)
+                    dnf_repo.enable()
+                    repos += list(dnf_repo)
+            else:
+                for repo in enable_repos:
+                    dnf_repo = base.repos.get_matching(repo)
+                    dnf_repo.enable()
+                    repos += list(dnf_repo)
+                for repo in disable_repos:
+                    dnf_repo = base.repos.get_matching(repo)
+                    dnf_repo.disable()
+                    repos += list(dnf_repo)
```

Comment:
The order of options is important, you can't apply all enable first and then all disable. For example `--disablerepo=* --enablerepo=something` is different than `--enablerepo=* --disablerepo=something`